lib/exercises.rb

```diff
@@ -1,20 +1,63 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(1)
+# Space Complexity: 0(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.empty?
+  return [strings] if strings.size == 1
+
+  hash1 = {}
+
+  # this is the original str
+  strings.each do |str|
+  
+    # this is the sorted str
+    word = str.split("").sort.join("")
+  
+    # check hash for word 
+    # if not exist, set str as hash value
+    if hash1[word].nil? 
+      hash1[word] = [str]
+    else
+      # or shovel str as hash value
+      hash1[word] << str
+    end
+  
+  end
+  return hash1.values
 end
 
+
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: 0(1)
+# Space Complexity: 0(n)
 def top_k_frequent_elements(list, k)
```

Comment:
This is most of the solution, the only weakness is that the `sort_by` you're doing doesn't preserve the order the elements were encountered in.  You also have the time complexity wrong because you have a loop through all the elements and you're sorting them.